azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/method_call.py

```diff
@@ -0,0 +1,39 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""This module contains a class representing method calls that are received and responded to.
+"""
+
+
+class MethodCall(object):
+    """Represents a call to a device or module method.
+
+    :ivar str name: The name of the method being called.
+    :ivar payload: The payload being sent with the call.
+    :ivar int response_timeout: The time in seconds before a response to the call is invalid.
+    """
+
+    def __init__(self, name, payload, response_timeout):
+        """Initializer for a MethodCall.
+
+        :param str name: The name of the method being called.
+        :param payload: The payload being sent with the call.
+        :param int response_timeout: The time in seconds before a response to the call is invalid.
+        """
+        self._name = name
+        self._payload = payload
+        self._response_timeout = response_timeout
+
+    @property
+    def name(self):
+        return self._name
```

Comment:
What is the value in making these properties instead of just attributes?

Reply:
The method call acts as a token to respond to the request with. Thus, it's semantically supposed to be a read-only object.

In truth the only field that's necessary, I believe, in order to respond is request ID (which I, of course, forgot to include on this structure >_>), but the idea is that this object works as a token, and the token should not be changed.

I do believe in trying to avoid enforcing read-only attributes whenever possible in Python, however, I think we can be a bit more cautious with user facing objects, especially ones that have semantic meaning like this one.

---
In reply to: [266089272](https://github.com/Azure/azure-iot-sdk-python-preview/pull/61#discussion_r266089272) [](ancestors = 266089272)

Reply:
sounds good (although i'd be careful with the use of the word "token" that has other meanings in the SDK. 

Also, process point: the commenter gets to close. if you don't want to change something, don't close it, wont-fix it instead please.

---
In reply to: [266114824](https://github.com/Azure/azure-iot-sdk-python-preview/pull/61#discussion_r266114824) [](ancestors = 266114824,266089272)